### DISCUSSION TITLE
Ethernet or WiFi over USB?

### DISCUSSION THREAD

**Discussion by adelie93** [rgerganov/gen5fw#20] [General]
My unit does not have WiFi built in. Anyone have an idea how to add WiFi over the USB port?

**Comment by al3emsy** [comment]
I saw your post yesterday, and while washing the car today I remembered something that might help, I remember having a USB stick that acted as a wifi module, you can find them on amazon they look a little something like this:
![image](https://github.com/rgerganov/gen5fw/assets/134560566/eb95ceec-c3b1-43c6-a617-60191b0e75d2)
there was also a dongle that connects to USB output and allows android auto and apple carplay to be used wirelessly, maybe you can play around and find a way to use it as a wifi adapter
![image](https://github.com/rgerganov/gen5fw/assets/134560566/41b1c211-d5f6-4c9a-b6ba-1be203f00a61)

maybe those would spark an idea or an approach, good luck!

**Comment by rgerganov** [comment]
Attaching USB WiFi adapter should be possible if there is a kernel module for that. You can refer to [this information](https://github.com/morrownr/USB-WiFi/blob/main/home/USB_WiFi_Chipsets.md) and try network adapters which are supported with 2.6+ and 3.0+ kernels. Building and loading the kernel module would be tricky though.
